### PR TITLE
[Merged by Bors] - Make CI friendlier

### PR DESCRIPTION
**Pull request by mockersf**
# Objective

- Make CI friendlier

## Solution

- CI now says hello to new contributor
- for some jobs with non obvious solutions to failures, give more context
  - example run should say which example failed
  - example doc should say the next action to do (add metadata or run the update script)
  - MSRV will say when it needs updating

I'm not completely sure everything is working and will try to trigger failures in this PR

### CODE REVIEW

**Comment by mockersf** [comment]
oh shoot, it's not possible to test in this repo for security reasons, which is why I had to add a new workflow 🤦 

I'll create a new repo to test

**Comment by mockersf** [comment]
Example of a comment left when the MSRV job fails: https://github.com/mockersf/test-actions/pull/3

Some security details:
* workflows triggered by `pull_request_target` are run on a PR with the workflow definition in main branch, not the one from the PR
* workflows triggered by `workflow_run` are run after another workflow, with their definition in main branch. They don't have access to the context that triggered the original workflow. I'm using the approach described [here](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/)

**Comment by mockersf** [comment]
Comments on the text of the messages are welcome!

File: .github/workflows/welcome.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+              }
+            }
+
+            await github.rest.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: `**Welcome**, new contributor!
+
+                Please make sure you're read our [contributing guide](CONTRIBUTING.md) and we look forward to reviewing your Pull request shortly ✨`
```
Comment (torsteingrindvik) [inline .github/workflows/welcome.yml:41]: Maybe something like an additional line:

"Thanks for helping Bevy grow! If you're not already there, we'd love to hear from you on our official Discord server <link here>, come say hi!"

Perhaps?

Comment (alice-i-cecile) [reply .github/workflows/welcome.yml:41]: Probably too noisy TBH :)

Comment (mockersf) [reply .github/workflows/welcome.yml:41]: I don't want to push discord that much. While it's active, it must not become mandatory for contributors...

**Review by Weibye — APPROVED** [review]
This is very helpful :)

**Review by alice-i-cecile — APPROVED** [review]
Very nice! The messages are nicely written.

bors r+

**Pull request closed without merging**